PROMPTS.md

```diff
@@ -0,0 +1,138 @@
+We create a folder with this name inside the top level directory of your plugin,
+and populate it with code templates.
+
+This module will also be added as the entry point to your plugin in `setup.cfg`. 
+This is how napari discovers plugins on launch.
+
+## short_description
+
+This should be a short description of what your plugin does. It will be listed
+in `setup.cfg` under the `description` field. If you publish your plugin to PyPI,
+this description will also be listed alongside your package name in search results.
+
+## include_reader_plugin
+
+Choosing `"y"` for this prompt will create an example reader hook implementation 
+inside your plugin's module in the file `_reader.py`. You can then edit the code in this
+file to achieve the reading functionality you want. For more information on
+readers see the [hook specification reference](https://napari.org/plugins/stable/hook_specifications.html#hook-specifications-reference).
+
+## include_writer_plugin
+
+Choosing `"y"` for this prompt will create an example writer hook implementation
+ inside your plugin's module in the file `_writer.py`. You can then edit the code in this file
+to achieve the writing functionality you want. For more information on writers see the [hook specification reference](https://napari.org/plugins/stable/hook_specifications.html#hook-specifications-reference).
+
+## include_dock_widget_plugin
+
+Choosing `"y"` for this prompt will create an example dock widget hook implementation
+ inside your plugin's module in the file `_dock_widget.py`. You can then edit the code in this file
+to achieve the dock widget functionality you want. For more information on dock widgets see the [hook specification reference](https://napari.org/plugins/stable/hook_specifications.html#hook-specifications-reference).
+
+## include_function_plugin
+
+Choosing `"y"` for this prompt will create an example function hook implementation
+ inside your plugin's module in the file `_function.py`. You can then edit the code in this file
+to achieve the functionality you want. For more information on function implementations see the [hook specification reference](https://napari.org/plugins/stable/hook_specifications.html#hook-specifications-reference).
+
+## use_git_tags_for_versioning
+
+The default for this prompt is `"n"`. If you choose `"n"` for this prompt, you will have to manually manage your version numbers
+when you create new releases of your plugin package. You can do
+this in `setup.cfg` under the `version` field. Choosing
+`"n"` at this prompt will add `version = 0.0.1` to your `setup.cfg`.
+
+If you choose `"y"` for this prompt, versions will be automatically
+managed by `setuptools_scm` when you make commits to your `git` repository.
+Note that in order to use this option, you must at the 
+very least run `git init` in your package's root directory. If you are not familiar
+with `git`, or cannot guarantee it will be installed on your users' machines,
```

Comment:
> or cannot guarantee it will be installed on your users' machines

this isn't really the issue, in the vast majority of cases, people will `pip install` a plugin in which case this works just fine regardless of whether the target has git. (when a wheel or sdist is built, the version gets put into `_version.py` and git is no longer required).  The _only_ case we've seen so far when this is an issue, is if someone wants to directly install a plugin directly from a git repo (for instance installing straight from github)...  I think that's a pretty rare use case and I personally still think that the default should be "y".  You could also point to the [instructions for deployment in the readme](https://github.com/napari/cookiecutter-napari-plugin#set-up-automatic-deployments) are relatively clear on how to deploy.

Reply:
I realized later we can actually do this in the cookiecutter. I do it for my personal cookiecutter, see here: https://github.com/tlambert03/pyrepo-cookiecutter/blob/f57f9d4cb47f96537a454bee691394443782095f/hooks/post_gen_project.py#L17-L20

Reply:
hmmm okay I think I'm only now starting to really understand the issue this prompt is solving. I agree with you that the default choice should be `"y"` because even as I'm trying to document this use case in the prompt reference I feel like it's so specific that maybe it shouldn't be there?